apps/speed.c

```diff
@@ -1163,6 +1178,48 @@ static int ECDH_EVP_derive_key_loop(void *args)
     return count;
 }
 
+static long eddsa_c[EdDSA_NUM][2];
+static int EdDSA_sign_loop(void *args)
+{
+    loopargs_t *tempargs = *(loopargs_t **) args;
+    unsigned char *buf = tempargs->buf;
+    EVP_MD_CTX **edctx = tempargs->eddsa_ctx;
+    unsigned char *eddsasig = tempargs->buf2;
+    unsigned int *eddsasiglen = &tempargs->siglen;
+    int ret, count;
+
+    for (count = 0; COND(eddsa_c[testnum][0]); count++) {
+        ret = EVP_DigestSign(edctx[testnum], eddsasig, (size_t *)eddsasiglen, buf, 20);
```

Comment:
Why the magic number?

Reply:
It's just a 'magic number' which is used by other comparable signature algorithms like DSA, ECDSA...Actually the 'buf' contains 36 bytes of random data initialized in advance and signature algorithms use that as the input of signing, so EdDSA continues that 'tradition' as well here...

Reply:
I think that 20 comes from when the default was SHA-1.
I commented on this at the end of https://github.com/openssl/openssl/pull/7073#issuecomment-417775946

Reply:
Ahh, I see